src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/PowershellScriptStep.java

```diff
@@ -29,18 +29,27 @@
 import org.jenkinsci.plugins.durabletask.DurableTask;
 import org.jenkinsci.plugins.durabletask.PowershellScript;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 
 /**
  * Asynchronous batch script execution.
  */
 public class PowershellScriptStep extends DurableTaskStep {
 
     private final String script;
+    private boolean stopOnError = false;
+
+    @DataBoundSetter public void setStopOnError(boolean stopOnError) {
+        this.stopOnError = stopOnError;
+    }
 
     @DataBoundConstructor public PowershellScriptStep(String script) {
         if (script == null) {
             throw new IllegalArgumentException();
         }
+        if(this.stopOnError){
+            script = "$ErrorActionPreference=\"Stop\"" + script;
+        }
```

Comment:
Does this do anything? The constructor will run before `setStopOnError` could be called, so I think this branch is never entered. That said, if your new tests pass, then maybe the existing behavior is already fine and this PR is not needed, or maybe your tests don't quite cover the right scenario.

Also more generally, I don't think you should modify the script like this here. I think the right way to do this would be to modify [powershellHelper.ps1](https://github.com/jenkinsci/durable-task-plugin/blob/0d250a6142908f591bda91d8658ab1b102611ce3/src/main/resources/org/jenkinsci/plugins/durabletask/powershellHelper.ps1) (or do something like [this](https://github.com/jenkinsci/durable-task-plugin/blob/0d250a6142908f591bda91d8658ab1b102611ce3/src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java#L152)) over in `durable-task` plugin and expose a new `@DataBoundSetter` on [`PowershellScript`](https://github.com/jenkinsci/durable-task-plugin/blob/0d250a6142908f591bda91d8658ab1b102611ce3/src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java).

Reply:
Good points. If I had to choose, I'd go with the [second option](https://github.com/jenkinsci/durable-task-plugin/blob/0d250a6142908f591bda91d8658ab1b102611ce3/src/main/java/org/jenkinsci/plugins/durabletask/PowershellScript.java#L152)
`powershellHelper.ps1` does not exist in the [binary version](https://github.com/jenkinsci/lib-durable-task) of `durabletask`, so you would lose support there.